File: timecapsule-database/src/main/java/com/timecapsule/infra/entity/Member.java
package com.timecapsule.infra.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@NoArgsConstructor
public class Member extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(unique = true)
    private String nickName;
    private String password;
    private String email;

    private Member(String nickName, String password, String email) {
        this.nickName = nickName;
        this.password = password;
        this.email = email;
    }

    public static Member of(String nickName, String password, String email) {
        return new Member(nickName, password, email);
    }

}

File: common/timecapsule-database/src/main/kotlin/com/timecapsule/database/entity/Member.kt
package com.timecapsule.database.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import java.lang.IllegalStateException
import javax.persistence.*

@Entity
data class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @Column(unique = true)
    val nickname: String,
    var password: String,
    @Column(unique = true)
    val email: String,
): BaseTimeEntity() {
    @get:JsonIgnore
    val nonNullId: Long
        get() = id ?: throw IllegalStateException("member id는 null일 수 없습니다.")
}
